cmd/soroban-rpc/internal/config/options.go

```diff
@@ -272,68 +273,128 @@ func (cfg *Config) options() ConfigOptions {
 			ConfigKey:    &cfg.RequestBacklogGetHealthQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-events-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-events-queue-limit"),
 			Usage:        "Maximum number of outstanding GetEvents requests",
 			ConfigKey:    &cfg.RequestBacklogGetEventsQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-network-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-network-queue-limit"),
 			Usage:        "Maximum number of outstanding GetNetwork requests",
 			ConfigKey:    &cfg.RequestBacklogGetNetworkQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-latest-ledger-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-latest-ledger-queue-limit"),
 			Usage:        "Maximum number of outstanding GetLatestsLedger requests",
 			ConfigKey:    &cfg.RequestBacklogGetLatestLedgerQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-ledger-entries-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-ledger-entries-queue-limit"),
 			Usage:        "Maximum number of outstanding GetLedgerEntries requests",
 			ConfigKey:    &cfg.RequestBacklogGetLedgerEntriesQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-get-transaction-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-get-transaction-queue-limit"),
 			Usage:        "Maximum number of outstanding GetTransaction requests",
 			ConfigKey:    &cfg.RequestBacklogGetTransactionQueueLimit,
 			DefaultValue: uint(1000),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-send-transaction-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-send-transaction-queue-limit"),
 			Usage:        "Maximum number of outstanding SendTransaction requests",
 			ConfigKey:    &cfg.RequestBacklogSendTransactionQueueLimit,
 			DefaultValue: uint(500),
 			Validate:     positive,
 		},
 		{
-			Name:         "request-backlog-simulate-transaction-queue-limit",
+			TomlKey:      strutils.KebabToConstantCase("request-backlog-simulate-transaction-queue-limit"),
 			Usage:        "Maximum number of outstanding SimulateTransaction requests",
 			ConfigKey:    &cfg.RequestBacklogSimulateTransactionQueueLimit,
 			DefaultValue: uint(100),
 			Validate:     positive,
 		},
+		{
+			TomlKey:      strutils.KebabToConstantCase("request-execution-warning-threshold"),
+			Usage:        "The request execution warning threshold is the predetermined maximum duration of time that a request can take to be processed before a warning would be generated",
```

Comment:
can this state where the warning is emitted, i.,e. is it a log or metric?

Reply:
sure. it emits it to both. ( for the metrics, it's just a counter ).